### ISSUE TITLE
Perception expander plugin (Improve peripheral vision)

### ISSUE THREAD

**Issue by apletnev** [koreader/koreader#2447]
* KOReader version: all
* Device: all

#### Issue

Based on that idea: [How to Triple Your Reading Speed in 20 Minutes (Tim Ferriss)](https://youtu.be/jeOHqI9SqOI?t=137) implemented the same approach (two vertical lines) for koreader. 

I've some disscussion about it with @houqp in #2397 so currently it is not work as a plugin. For more info read #2397
Guys, take a look at the video, and provide your feedback e.q. plugin name, settings, etc.

![perceptualexpander](https://cloud.githubusercontent.com/assets/220872/21399321/c6fb06b0-c7b3-11e6-8140-bd0b5c3b1301.gif)

**Comment by AlanSP1** [comment]
This looks like great idea. I like Ferris's books, learned a lot from him before.

Main thing is, people will need to learn what this plugin is all about, how to use it. We will need instructions for that (and for many other things).

But, I like it and I would like to have this included and being able to use it. And learn how to use it properly.

**Comment by houqp** [comment]
I like this feature too, good work :) Are you using the alphacontainer to draw those lines? If not, I recommend using it. The name speed read might be too generic. `Perception expander` is actually not a bad name :P I hope we can have more tools in the future to help people train their reading speed.

You don't have to implement it as a plugin for now. I can help refactor it into a plugin later.

**Comment by KenMaltby** [comment]
"Speed Reading Trainer"  sounds practical.   "Perception expander" sounds philosophical.  I prefer the practical,

**Comment by houqp** [comment]
What happens when we add other speed reading trainer plugins in the future? How about adding perception expander as a submenu under speed reading trainer menu?

**Comment by retrue** [comment]
Imho, adding perception expander as a submenu would make sense only and when there are other speed reading trainer plugins for KOReader.
If it is added in a submenu now, it would only add another step to reach the plugin, without any advantage.

**Comment by houqp** [comment]
If number of clicks is a concern here, how about naming the menu entry as `Speed reading module - perception expander`? We have more than enough space to hold this string in the menu.

Giving this plugin a generic name such as `speed reading trainer` will cause two problems down the road if we are going to add more speed reading training features in the future:

* inconsistent UX, user who are already familiar with the generic name now need to learn that the old module is renamed to `perception expander` and the old name is a new top level menu entry with a sub menu.
* requires code refactoring, now all reference to speed reading trainer needs to be updated with the new name

**Comment by houqp** [comment]
@apletnev I have made the readerview module pluggable in PR https://github.com/koreader/koreader/pull/2454. Now you should be able to implement everything in a plugin. All you have to do is put this into the init method:

```lua
function PerceptionExpander:init()
    self.view:registerViewModule("perception_expander", self)
end

function PerceptionExpander:resetLayout()
    -- screen resized, update width and height accordingly
end
```

**Comment by apletnev** [comment]
@houqp Thank you. going to rewrite as plugin module. Will register it in menu as "Speed reading module - perception expander"

**Comment by poire-z** [comment]
A small thing:
I think `PerceptionExpander:onSaveSettings()` ( https://github.com/koreader/koreader/commit/2bc1be995616022a2408771fad4a5b3255d5fe33#diff-8a9416d5c212fd39ee25121820043bb8R193 ) is unecessary.
You're already doing all the explicite self:saveSettings() needed whenever a setting is changed.
This onSaveSettings() makes the file being rewritten unchanged on each suspend/exit, and created even if people have never used the plugin.

**Comment by houqp** [comment]
Agree, feel free to submit a patch.

**Comment by Frenzie** [comment]
Implemented plus the issue pointed out by @poire-z was fixed in #2487, so I'm closing this now.
